### ISSUE TITLE
Some Questions about Aspire 

### ISSUE THREAD

**Issue by LianwMS** [dotnet/aspire#1011]
Hi Team,

I want to express my appreciation for the work on the Aspire project. I am trying the sample project and learning from the [doc](https://learn.microsoft.com/en-us/dotnet/aspire/get-started/aspire-overview). 

But I still have some questions about the Asprie, mainly focus on service discovery and health check. The following is some of my understanding and questions:

1. My understanding is both service discovery and health check are based on the functionality on .net. Aspire seems add a convenient layer on ASP.NET Core library. 

2. In health check, it will expose two APIs: /health and /alive... I cannot understand the different clearly. Is it possible that if the endpoint return healthy for /health but unhealthy for /alive?

3. In the Service Discovery part, it also using the existing Asp.Net core functionality. I suppose it will not consume health check result from endpoint by default. If I want to consume the /health, I need customize by myself.

**Comment by davidfowl** [comment]
> My understanding is both service discovery and health check are based on the functionality on .net. Aspire seems add a convenient layer on ASP.NET Core library.

Yes. It uses the built in primitives and puts them together in a way that provides an end to end opinionated experience.

> In health check, it will expose two APIs: /health and /alive... I cannot understand the different clearly. Is it possible that if the endpoint return healthy for /health but unhealthy for /alive?


health is about getting the health status of a service and all of its dependencies. This can be mapped to a readiness probe in Kubernetes. The alive endpoint tells you if the service itself is healthy irrespective of its dependencies (can you even reply to a simple http request?). This can be mapped to a liveness probe in Kubernetes.

> In the Service Discovery part, it also using the existing Asp.Net core functionality. I suppose it will not consume health check result from endpoint by default. If I want to consume the /health, I need customize by myself.

Service discovery is being added to Microsoft.Extensions. It doesn’t do any health checking today. They are unrelated subsystems. So yes, you’d need to customize yourself.
